### DISCUSSION TITLE
using JFrog Artifactory instead Nodejs.org registry

### DISCUSSION THREAD

**Discussion by Amosc91** [coreybutler/nvm-windows#999] [Support] [answered]
Hello, I have an on-premise network that can’t access the internet, and I have JFrog Artifactory that I am trying to deploy the Nodejs zip installations files and I can’t make it work.

I've configured the nvm node_mirror to the url of my repository in the artifactory  

How should I arrange the structure of the artifactory repository ?
I tried a paths like "v17.9.1/node-v17.9.1-win-x64.zip" or a folder without zip of "node-v17.9.1-win-x64" 
even tried "  v17.9.1/node-v17.9.1-win-x64" without zip
when running the command "nvm install 17" it can't find it.
Is it even possible using artifactory instead of the nodejs registry? 
hope to find solutions here, thanks advance

**Comment by coreybutler** [comment, marked as answer]
You should be able to mirror the Node distribution directory found at https://nodejs.org/dist/. Notice each version is in its own directory using the syntax `vX.X.X`. NVM4W also relies on the `index.json` file found in this directory.

Npm also needs to be mirrored if you want it to be available for users. New versions of npm come from https://github.com/npm/cli/releases. Npm changed their repo quite a while ago. If older versions are required, files from https://github.com/npm/npm/releases will be needed as well.

**Reply by Amosc91** [reply]
Thanks it worked after I added  the "latest" folder 🙏

**Reply by zuzusik** [reply]
@coreybutler any hints on how to make artifactory not to cache index.json and index.tab? By default artifactory caches them and so version list locks in.

**Reply by coreybutler** [reply]
@zuzusik - Unfortunately, I think you'd have to talk to JFrog about that. Only the server can clear out the cache. That's pretty perplexing. I'd be surprised if JFrog doesn't have some sort of solution for that.

**Reply by zuzusik** [reply]
Thanks @coreybutler ! 

Yeah, we'd need to talk to artifactory about that. So far I don't really see any options to fine tune cache per file except this now deprecated plugin here: https://github.com/jfrog/artifactory-user-plugins/tree/master/deprecated-plugins/metadata/expireFilesMetadata

**Reply by NicklasWallgren** [reply]
> Thanks @coreybutler !
> 
> Yeah, we'd need to talk to artifactory about that. So far I don't really see any options to fine tune cache per file except this now deprecated plugin here: https://github.com/jfrog/artifactory-user-plugins/tree/master/deprecated-plugins/metadata/expireFilesMetadata

We're experiencing the same issue. Did you find a solution besides manually deleting the index.json and index.tab in Artifactory?
